state/application.go

```diff
@@ -356,10 +356,37 @@ func (op *DestroyApplicationOperation) destroyOps() ([]txn.Op, error) {
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		if n != 0 {
-			return nil, errors.Errorf("application is used by %d offer%s", n, plural(n))
+		// TODO(tsm) check offers, if there are no relations/connections, proceed
+		if n == 0 {
+			ops = append(ops, countOp)
+		} else {
+			offers, err := applicationOffersDocs(op.app.st, op.app.Name())
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			for _, offer := range offers {
+				// TODO(tsm) if there are connections, error out
+				connections, err := op.app.st.OfferConnections(offer.OfferUUID)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+
+				if len(connections) > 0 {
+					return nil, errors.Errorf("application is used by an offer with current relation%s", n, plural(len(connections)))
```

Comment:
We just want to preserve the original (overarching) error. This will bail early without reporting the number of offers for the app